build.sbt

```diff
@@ -43,6 +43,22 @@ def conditionalDependsOn(prj: Project): Project = {
   }
 }
 
+/**
+  * It has been a struggle for us to override settings in subprojects.
+  * An example would be adding a dependency to rocketchip on midas's targetutils library,
+  * or replacing dsptools's maven dependency on chisel with the local chisel project.
+  *
+  * This function works around this by specifying the project's root at src/ and overriding
+  * scalaSource and resourceDirectory.
+  */
+def freshProject(name: String, dir: File): Project = {
```

Comment:
I feel like we've tried this before, and it hasn't completely worked for all use-cases. Maybe @davidbiancolin has some additional insights?

Reply:
No, this is just a method wrapping around the gloriously hacky hack that we used for rocket-chip. LGTM